admin/resources/js/components/form/Select.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from "react";
+import { Control, useController } from "react-hook-form";
+
+interface SelectProps extends React.SelectHTMLAttributes<HTMLSelectElement> {
+  /** Object that connects controlled components to react hook form. */
+  control: Control<any>;
+  /** The text for label associated with select input. */
+  label: string;
+  /** A string specifying a name for the input control. */
+  name: string;
+  /** List of options for the select element. */
+  options: { value: string | number; text: string }[];
+  /** Object set of validation rules to impose on input. */
+  rules?: { [key: string]: any };
+  /** List of validation messages that correspond with the rules list. */
+  ruleMessages?: { [key: string]: string };
+}
+
+const Select: React.FunctionComponent<SelectProps> = ({
+  control,
+  label,
+  name,
+  options,
+  rules,
+  ruleMessages,
+  ...rest
+}) => {
+  const { field, fieldState } = useController({
+    control,
+    name,
+    rules,
+  });
+  return (
+    <div>
+      <label htmlFor={name}>{label}</label>
```

Comment:
As @tristan-orourke [mentioned for the Input component](https://github.com/GCTC-NTGC/gc-digital-talent/pull/73/files#r650197744), the same applies for the Select component about the need for htmlFor to match the id of the input.